### PR TITLE
Adaptive launcher icon

### LINKED ISSUE #82: Adaptive launcher icon
> Android 8.0 (API level 26) introduces adaptive launcher icons, which can display a variety of shapes across different device models. - [Source](https://developer.android.com/guide/practices/ui_guidelines/icon_design_adaptive)

- [Understanding Android adaptive icons](https://medium.com/google-design/understanding-android-adaptive-icons-cee8a9de93e2)
- [Designing adaptive icons](https://medium.com/google-design/designing-adaptive-icons-515af294c783)
- [Implementing adaptive icons](https://medium.com/google-developers/implementing-adaptive-icons-1e4d1795470e)
- [VectorDrawable adaptive Icons](https://medium.com/androiddevelopers/vectordrawable-adaptive-icons-3fed3d3205b5)
- [Adapticon](https://adapticon.tooo.io)
- [AdaptiveIconPlayground](https://github.com/nickbutcher/AdaptiveIconPlayground)

The rendering of the launcher icon changes for different Android versions once the `targetSdkVersion` is set to `26`.

### Android 6, targetSdkVersion 25 or 26
34c3-android-9-25

### Android 9, targetSdkVersion 25
34c3-android-9-25

### Android 9, targetSdkVersion 26
34c3-android-9-26

# Related issues
- [Issue #58: Use targetSdkVersion 25 (Nougat)](https://github.com/EventFahrplan/EventFahrplan/issues/58)
- [Issue #59: Use targetSdkVersion 26 (Oreo)](https://github.com/EventFahrplan/EventFahrplan/issues/59)

### PR DESCRIPTION
**Pull request by AndyScherzinger**
This is my first PR to this project, hope you like it. Please let me know if anything needs to be changed.

Resolve #82  - Adding an adaptive launcher icon

**With safe zone shown**
2018-11-19 16_59_18-asset studio

**safe zone hidden**
2018-11-19 16_59_35-asset studio

I used https://github.com/EventFahrplan/EventFahrplan/files/2457091/34C3_WHITE_two_rows.svg.zip for the foreground and the given color codes for the background (see `assets/background.svg` added to this PR), while the foreground is slightly resized to fit within the safe zone (see first screenshot).

I also added the round icons (Android v25) which aren't "active" at the moment since that would need a change in the `AndroidManifest.xml` which then makes round icons mandatory for any of the builds so all builds need to provide round launcher icons too. I can remove the images from the PR, just need a ping.

### CODE REVIEW

File: assets/background.svg [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (johnjohndoe) [inline assets/background.svg]: Can you please rename the file to `launcher-background-34c3.png` so it is easier to associate it with its product flavor?

Comment (AndyScherzinger) [reply assets/background.svg]: sure thing!

**Comment by AndyScherzinger** [comment]
@johnjohndoe 
1. I'll make the round icon the "default" one
2. I'll try to match the colors (while I thought I did do that based on the color code given in the issue ticket)
3. I'll reformat the xmls, no problem :)

I can also do a rebase, wouldn't be an issue but can also check the ammend command (normally I always did rebases, never used ammend until now :)

**Comment by johnjohndoe** [comment]
@AndyScherzinger 

2. Sorry for the wrong value. It seems that the lower brighter red tone in the old icon is actually not a solid color but a gradient. I picked random pixels in that area and they return different hex values.

With regards to the size of the foreground layer you might want to play around with [this version of Android Asset Studio](https://github.com/johnjohndoe/AndroidAssetStudio/tree/launcher-icon-size) which I used in the past to let the icon be generated a bit larger. I don't think this will help with the XML icon, though. Seems like @romannurik's version does not support generating XMLs.

Rebase is also fine. Just update your local `master` before. Please don't merge `master` into your branch.

**Comment by AndyScherzinger** [comment]
@johnjohndoe I set the round icon as the default while I generated everything within Android Studio. 
I checked the Asset Studio links you posted but didn't see a way to use anything else than a simple color for the background. I can make the background xml have gradients, I'd just need to knwo which hex values I should use.

**Comment by johnjohndoe** [comment]
You can use the colors which are defined in the SVG. I believe the gradient came from  the Asset Studio wizard in Android Studio. But please do not invest too much time. The life time of the 34c3 app is already determined to be less then a month since 35c3 is near.

**Comment by AndyScherzinger** [comment]
@johnjohndoe no worries :) Can you post a link to the svg which has the actual background gradients? I couldn't find it in the actual master.

**Comment by johnjohndoe** [comment]
@AndyScherzinger Sorry, my mistake. I thought the SVG linked [here](https://github.com/EventFahrplan/EventFahrplan/issues/82#issuecomment-427926736) would contain the colors - but is just white. Therefore, feel free to improvise based on the [old logo](https://github.com/EventFahrplan/EventFahrplan/pull/86#pullrequestreview-177724302).

**Comment by AndyScherzinger** [comment]
@johnjohndoe Done :) I also put everything in one commit now to get rid of the try-and-fail commit to get the gradients right.

**Comment by johnjohndoe** [comment]
@AndyScherzinger That looks really nice. Thank you.

Almost perfect - I just fixed a few minor things and force pushed the branch to avoid another ping-pong round.
- Remove `app/src/ccc34c3/res/mipmap-anydpi-v26/ic_launcher_round.xml` since `ic_launcher.xml` already contains the round icon.
- Update `app/src/main/ic_launcher-web.png` with a round icon.
- Remove the former store icon `app/src/ccc34c3/web_hi_res_512.png`.
- Add `assets/launcher-foreground-34c3.svg` to have it right next to `assets/launcher-background-34c3.svg`.
- A bit of file formatting.
- Revised the commit message.

**Comment by AndyScherzinger** [comment]
Sweet! Glad I could be of help ❤️

**Pull request merged**